cassandra/cluster.py

```diff
@@ -4321,14 +4321,17 @@ def was_applied(self):
         """
         For LWT results, returns whether the transaction was applied.
 
-        Result is indeterminate if called on a result that was not an LWT request.
+        Result is indeterminate if called on a result that was not an LWT request or on
+        a :class:`.query.BatchStatement` containing LWT. In the latter case either all the batch
+        succeeds or fails.
 
-        Only valid when one of tne of the internal row factories is in use.
+        Only valid when one of the of the internal row factories is in use.
         """
         if self.response_future.row_factory not in (named_tuple_factory, dict_factory, tuple_factory):
             raise RuntimeError("Cannot determine LWT result with row factory %s" % (self.response_future.row_factsory,))
-        if len(self.current_rows) != 1:
```

Comment:
I finally removed this because it was failing for the batch test. We can have a query string with a batch were more than one row fails.

Reply:
As we discussed last week, I think we should avoid any runtime behavior change. I think that it is fine to assume the user *has* to use a BatchStatement and not a query string if he wants this was_applied to work. We at least provide a small improvement for that case.

We can also create a follow-up ticket about this to see what we should change in the next major release.

Reply:
> I think that it is fine to assume the user has to use a BatchStatement and not a query string

I'd think as well, unfortunately however in CQLEngine we are using a string for the batch statement so [this assert](https://github.com/datastax/python-driver/pull/912/files#diff-8d5a882a817cdaaeeb0ce18fb53ef274R158) was failing in the test, this was the reason I changed the logic in was_applied. I think this is a bug in CQLEngine now since the user may wrongfully believe the batch statement was applied but we can handle this in a follow up ticket as well.

If you are +1 I'll do the following: leave the logic originally agreed in was_applied, create a bug ticket for CQLEngine and a follow up ticket for the major release.